cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -150,31 +151,42 @@ public void testInvalidAppWithDuplicateDatasets() throws Exception {
     AppRequest<AppWithDuplicateData.ConfigClass> createRequest = new AppRequest<>(
       new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()),
       new AppWithDuplicateData.ConfigClass(false, false, false));
-    deployApplication(appId, createRequest);
+    try {
+      deployApplication(appId, createRequest);
+    } finally {
+      deleteApplication(appId);
```

Comment:
so we call stop for all applications in TestBase#afterTest, if the app was deleted, then stop would fail. currently in your test case, the deploy application method doesn't add the application manager to the applicationManagers list in TestBase, but there are other deployApplication methods which do. so there are some inconsistencies.